File: app/src/main/java/com/security/passwordmanager/Pair.java
package com.security.passwordmanager;

public class Pair<X, Y> {

    private X first;
    private Y second;

    public Pair(X first, Y second) {
        setFirst(first);
        setSecond(second);
    }

    public Pair() {
        this(null, null);
    }

    public X first() {
        return first;
    }

    public Y second() {
        return second;
    }

    public void setFirst(X first) {
        this.first = first;
    }

    public void setSecond(Y second) {
        this.second = second;
    }

    public boolean isFirst(X first) {
        return this.first.equals(first);
    }

    public boolean isSecond(Y second) {
        return this.second.equals(second);
    }

    public boolean equals(Pair<X, Y> pair, Integer position) {
        if (position == 0)
            return isFirst(pair.first);
        else if (position == 1)
            return isSecond(pair.second);
        else
            return isFirst(pair.first) && isSecond(pair.second);
    }
}

File: app/src/main/java/com/security/passwordmanager/Pair.kt
package com.security.passwordmanager

class Pair<X, Y>(var first: X, var second: Y)

operator fun <A>Pair<A, A>.get(index : Int): A = if (index == 0) first
else second

operator fun <A>Pair<A, A>.set(index: Int, value : A) = if (index == 0) first = value
else second = value

fun <A>Pair<A, A>.setAll(block: (element: A) -> Unit) {
    block(first)
    block(second)
}

fun <B>getPair(block: (position: Int) -> B) =
    Pair(block(0), block(1))